### PR TITLE
Supports autopkgtests now on artful+

### PR DESCRIPTION
**Pull request by didrocks**
* swith from lightdm to gdm for loging manager autologin
* wait on 2 gnome-session process to run to start tests
* report correct log files

### CODE REVIEW

File: debian/tests/setup-testbed [hunk 1]
```diff
@@ -37,12 +37,10 @@ Section "Device"
...
-    sudo -n mkdir -p /etc/lightdm/lightdm.conf.d
-    cat << EOF | sudo -n tee /etc/lightdm/lightdm.conf.d/umaketests.conf
-[SeatDefaults]
-autologin-user=ubuntu
+# enable autologin to get user session started
+cat << EOF | sudo -n tee -a /etc/gdm3/custom.conf
+AutomaticLoginEnable = true
+AutomaticLogin = ubuntu
 EOF
 fi
```
Comment (LyzardKing) [inline debian/tests/setup-testbed]: This `fi` needs to be removed...
Then the modification works with my changes to runtests!

**Comment by didrocks** [comment]
Thanks for looking ;)
Forced push! Happy that you can now run the tests, especially after doing this change blindly ;)

File: debian/tests/utils [hunk 2]
```diff
@@ -74,22 +74,21 @@ function run_tests {
...
 }
 
-# wait for compiz to start, exit 1 after a timeout if not running.
-function wait_for_compiz {
+# wait for user session to start, exit 1 after a timeout if not running.
+function wait_for_user_session {
     timeout=100
-    while [ ! `pgrep -c compiz` -gt 0 ]; do
+    # we want 2 gnome-sessions: one for gdm, one for the user session
+    while [ ! `pgrep -c gnome-session` -eq 2 ]; do
```
Comment (LyzardKing) [inline debian/tests/utils]: This does need to be set to 1 to work.

**Comment by didrocks** [comment]
Fixed and merged! Thanks :)

**Pull request merged**